diff/diff_test.go

```diff
@@ -47,7 +59,7 @@ func TestParseHunkNoChunksize(t *testing.T) {
 	}
 	h := diff[0]
 	h.Body = nil // We're not testing the body.
-	if !reflect.DeepEqual(h, correct) {
+	if !cmp.Equal(h, correct) {
```

Comment:
minor: if switching to `go-cmp`, it would be nice to use `cmp.Diff` instead so we get more readable error output.

Reply:
Optional comment.

The current version is correct. It's possible to simplify slightly by reusing `cmp.Diff` output both as the test condition and the human-friendly test failure output:

```Go
if diff := cmp.Diff(correct, h); diff != "" {
	t.Errorf("%s: mismatch (-want +got):\n%s", diff)
}
```

It can also be helpful to make `cmp.Diff` output more consistent across different Go codebases by placing "want"—in this case `correct`—first, and "got"—here `h`—second, then the output style is always "-want +got".

(This is based on an example in go-cmp documentation, https://pkg.go.dev/github.com/google/go-cmp/cmp?tab=doc#example-Diff-Testing.)

Reply:
yeah this pattern is nice and we (mostly) consistently do that in our other repos, but not a blocker for this PR.